pkg/nodediscovery/nodediscovery.go

```diff
@@ -454,8 +454,11 @@ func (n *NodeDiscovery) mutateNodeResource(nodeResource *ciliumv2.CiliumNode) er
 
 	nodeResource.ObjectMeta.Labels = k8sNodeParsed.Labels
 
-	localCN := n.localNode.ToCiliumNode()
-	nodeResource.ObjectMeta.Annotations = localCN.Annotations
+	// This is for syncing relevant node annotations from the k8s node to the
+	// CiliumNode object
+	ln := n.localNode.DeepCopy()
+	ln.Annotations = k8sNodeParsed.Annotations
+	nodeResource.ObjectMeta.Annotations = ln.GetCiliumAnnotations()
```

Comment:
I think we can get rid of the deep copy with:

`nodeResource.ObjectMeta.Annotations = k8sNodeParsed.GetCiliumAnnotations()`

On a side note, unrelated to this PR, I see that the `types` package is imported twice:

```
	"github.com/cilium/cilium/pkg/node/types"
	nodeTypes "github.com/cilium/cilium/pkg/node/types"
```

So we can get rid of one of those (but let's leave that for a follow up PR, no need to do this here).